common/src/main/java/io/netty/util/concurrent/DefaultEventExecutorChooserFactory.java

```diff
@@ -43,6 +46,19 @@ private static boolean isPowerOfTwo(int val) {
         return (val & -val) == val;
     }
 
+    private static final class SingleEventExecutorChooser implements EventExecutorChooser {
```

Comment:
@normanmaurer @njhill 
On Netty 5 these 3 classes could be collapsed into lambdas in order to let the JVM trust non static final fields (see http://hg.openjdk.java.net/jdk/jdk/file/2c1af559e922/src/hotspot/share/ci/ciField.cpp#l203) eg executors array. 
There is a good chance that it could be just implemented with:
```java
@Override
    public EventExecutorChooser newChooser(final EventExecutor[] executors) {
        final int executorsLength = executors.length;
        final AtomicInteger idx = new AtomicInteger();
        if (executorsLength > 0) {
            return () -> {
                return executors[Math.abs(idx.getAndIncrement() % executorsLength)];
            };
        } else {
            final EventExecutor executor = executors[0];
            return () -> {
                return executor;
            }
        }
    }
```